portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -23,42 +23,80 @@
+            comments.add(getCommentFromCommentEntity(entity));
+        }
+        response.setContentType("application/json;");
+        response.getWriter().println(convertToJsonUsingGson(comments));
+    }
+
     private Comment getCommentFromRequest(HttpServletRequest request){
         String firstName = request.getParameter("first-name");
         String lastName = request.getParameter("last-name");
         String email = request.getParameter("email");
         String phone = request.getParameter("phone");
         String message = request.getParameter("comment");
-        Comment newComment = new Comment(firstName, lastName, email, phone, message);
+        String jobTitle = null;
         if(request.getParameterValues("type")!=null){ //box is checked
-            String jobTitle = request.getParameter("job-title");
-            newComment.addJobTitle(jobTitle);
+            jobTitle = request.getParameter("job-title");
         }
+        Comment newComment = new Comment(firstName, lastName, email, phone, message, jobTitle);
         return newComment;
     }
-  
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{ 
-        response.setContentType("application/json;");
-        response.getWriter().println(convertToJsonUsingGson(comments));
+
+    private void putCommentToDatastore(Comment myComment){
+        Entity commentEntity = getCommentEntityFromComment(myComment);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+    }
+
+    private Entity getCommentEntityFromComment(Comment myComment){
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("firstName", myComment.getFirstName());
+        commentEntity.setProperty("lastName", myComment.getLastName());
+        commentEntity.setProperty("email", myComment.getEmail());
+        commentEntity.setProperty("phone", myComment.getPhone());
+        commentEntity.setProperty("message", myComment.getMessage());
+        commentEntity.setProperty("jobTitle", myComment.getJobTitle());
+        return commentEntity;
+    }
+
+    private PreparedQuery getCommentsFromDatastore(){
+        Query commentsQuery = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(commentsQuery);
+        return results;
+    }
+
+    private Comment getCommentFromCommentEntity(Entity commentEntity){
+        String firstName = (String) commentEntity.getProperty("firstName");
+        String lastName = (String) commentEntity.getProperty("lastName");
+        String email = (String) commentEntity.getProperty("email");
+        String phone = (String) commentEntity.getProperty("phone");
+        String message = (String) commentEntity.getProperty("message");
+        String jobTitle = (String) commentEntity.getProperty("jobTitle");
```

Comment:
Since the entity property names are used in two places, we could extract them to private constants. That way they are defined in one place and it is less error prone.

Reply:
done